stubs/untangle/untangle.pyi

```diff
@@ -0,0 +1,37 @@
+from _typeshed import Self
+from collections.abc import Iterator, Mapping
+from typing import Any
+from xml.sax import handler
+
+def is_string(x: object) -> bool: ...
+
+class Element:
+    children: list[Element]
+    is_root: bool
+    cdata: str
+    def __init__(self, name: str | None, attributes: Mapping[str, Any] | None) -> None: ...
+    def add_child(self, element: Element) -> None: ...
+    def add_cdata(self, cdata: str) -> None: ...
+    def get_attribute(self, key: str) -> Any | None: ...
+    def get_elements(self, name: str | None = ...) -> list[Element]: ...
+    def __getitem__(self, key: str) -> Any | None: ...
+    def __getattr__(self, key: str) -> Element: ...
```

Comment:
Just so you're aware: type checkers have special handling for `__getattr__` methods. If they see one defined, they'll assume that something funky is going on, and will generally assume any arbitrary attribute access is okay. E.g. because__getattr__ is included in the stub here, they won't emit an error for the following:

```python
e = Element("foo", {})
e.non_existing_attribute
```

That might be desirable in this case — but sometimes it's useful to omit `__getattr__` methods from stubs, even if they exist at runtime, to avoid this behaviour. Your call.

Reply:
Good to know. In this case I think it's ok, since the library returns an object with arbitrary attributes based on the structure of the XML file being parsed. So from a type checking perspective I think arbitrary access _is_ ok. If you try and access an attribute that isn't there in the XML, the library raises an `AttributeError`, but it's entirely dependent on the XML being parsed.